platform/viewer/src/connectedComponents/ConnectedUserPreferencesModal.js

```diff
@@ -0,0 +1,59 @@
+import { connect } from 'react-redux';
+import { UserPreferencesModal } from '@ohif/ui';
+import OHIF from '@ohif/core';
+import { hotkeysManager } from './../App.js';
+import cloneDeep from 'lodash.clonedeep';
+
+const { setUserPreferences } = OHIF.redux.actions;
+
+const mapStateToProps = (state, ownProps) => {
+  const isEmpty = obj => Object.keys(obj).length === 0;
+  const newHotKeysData =
+    state.preferences && !isEmpty(state.preferences.hotKeysData)
+      ? state.preferences.hotKeysData
+      : hotkeysManager.hotkeyDefinitions;
+  hotkeysManager.setHotkeys(hotkeysManagerFormatter(cloneDeep(newHotKeysData)));
+  return {
+    isOpen: ownProps.isOpen,
+    windowLevelData: state.preferences ? state.preferences.windowLevelData : {},
+    hotKeysData: newHotKeysData,
+  };
+};
+
+/* HotkeysManager's hotkeysDefinitions have different output/input. */
+const hotkeysManagerFormatter = hotKeysData => {
+  const hotKeysCommands = Object.keys(hotKeysData);
+  return hotKeysCommands.map(commandName => {
+    const definition = hotKeysData[commandName];
+
+    return {
+      commandName,
+      keys: definition.keys,
+      label: definition.label,
+    };
+  });
+};
+
+const mapDispatchToProps = (dispatch, ownProps) => {
+  return {
+    onSave: ({ windowLevelData, hotKeysData }) => {
+      hotkeysManager.setHotkeys(
+        hotkeysManagerFormatter(cloneDeep(hotKeysData))
+      );
+      dispatch(setUserPreferences({ windowLevelData, hotKeysData }));
```

Comment:
Considering you are closing modal anyway (assuming this is the right approach), can you invert the other? first you close then you dispatch to store (this kind of prevents user of dispatching twice). Same bellow

Reply:
Done! thx.